bin/pycbc_compress_bank

```diff
@@ -40,72 +40,105 @@ from pycbc import filter
-                help="The interpolation to use for decompressing the "
-                     "waveforms for checking tolerance. Options are 'linear', "
-                     "or any interpolation recognized by scipy's interp1d "
-                     "kind argument. Default is linear.")
+                    help="The maximum mismatch to allow between the "
+                    "interpolated waveform and the full waveform. Points"
+                    "will be added to the compressed waveform until its "
+                    "interpolation has a mismatch <= this value."
+                    " Default is 0.001.")
+parser.add_argument("--interpolation", type=str,
+                    default="inline_linear",
+                    help="The interpolation to use for decompressing the "
+                    "waveforms for checking tolerance. Options are "
+                    "'inline_linear', which assumes that the "
+                    "frequency samples of the amplitude and phase "
+                    "interpolants are the same, or any interpolation "
+                    "recognized by scipy's interp1d kind argument, which "
+                    "allows the frequency interpolant samples to be "
+                    "different for the amplitude and phase. "
+                    "Default is inline_linear.")
 parser.add_argument("--precision", type=str, choices=["double", "single"],
-                default="double",
-                help="What precision to generate and store the waveforms with;"
-                     " default is double.")
+                    default="double",
+                    help="What precision to generate and store the"
+                    "waveforms with; default is double.")
 parser.add_argument("--force", action="store_true", default=False,
                     help="Overwrite the given hdf file if it exists. "
-                         "Otherwise, an error is raised.")
+                    "Otherwise, an error is raised.")
 parser.add_argument("--verbose", action="store_true", default=False)
 
 
 args = parser.parse_args()
 
 pycbc.init_logging(args.verbose)
 
+num_compression_algorithms = len(args.compression_algorithm)
+
+if num_compression_algorithms == 1:
+    compression_type = args.compression_algorithm[0]
+    if compression_type is 'partial_compress_rom':
+        raise ValueError("""partial_compress_rom cannot be used as a
+                         compression algorithm for TaylorF2 waveforms.
+                         Specify a different compression algorithm
+                         compatible with TaylorF2 and
+                         SEOBNRv2_ROM_DoubleSpin.""")
+
+elif num_compression_algorithms == 2:
+    compression_type_approximant1, compression_type_approximant2 = args.compression_algorithm
+    compression_type1, approximant1 = compression_type_approximant1.split(':')
+    compression_type2, approximant2 = compression_type_approximant2.split(':')
+
```

Comment:
As I mentioned above, don't hard code the number of arguments to be 1 or 2. You can deal with the arbitrary case by creating a dictionary mapping approximant name -> compression algorithm. To ensure that the command encompasses all of the approximants in the bank, move this parsing to after the bank is loaded, so you can get all the approximants present in the bank. Then, when done parsing the `compression_alrgorithm` option, check that all of the approximants are covered, raising an error otherwise.